clients/src/test/java/org/apache/kafka/common/protocol/ErrorsTest.java

```diff
@@ -60,4 +61,14 @@ public void testNoneException() {
         assertNull("The NONE error should not have an exception", Errors.NONE.exception());
     }
 
+    @Test
+    public void testForExceptionInheritance() {
+        class ExtendedTimeoutException extends TimeoutException {}
```

Comment:
@becketqin They all won't and shouldn't. Almost all of the exceptions were migrated to client exceptions completely. The only exception that wasn't completely migrated was InvalidMessageException. This is because it is thrown/used in both client and server code. Instead it was changed to inherit from CorruptRecordException so that it could still be mapped while also maintaining client compatibility. This patch should fix that actual use case though, since forException didn't respect superclasses before.

If you think its important, I am happy to add test to core that ensures InvalidMessageException maps correctly with Errors.forException.